cmd/config/set.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Authors of Hubble
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"encoding/csv"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cast"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func newSetCommand(vp *viper.Viper) *cobra.Command {
+	return &cobra.Command{
+		Use:   "set KEY [VALUE]",
+		Short: "Set an individual value in the hubble config file",
+		Long: "Set an individual value in the hubble config file.\n" +
+			"If VALUE is not provided, the value is reset to its default value.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var val string
+			switch len(args) {
+			case 2:
+				val = args[1]
+				fallthrough
+			case 1:
+				return runSet(cmd, vp, args[0], val)
+			default:
+				return fmt.Errorf("invalid arguments: set requires exactly 1 or 2 argument(s): got '%s'", strings.Join(args, " "))
+			}
+		},
+	}
+}
+
+func runSet(cmd *cobra.Command, vp *viper.Viper, key, value string) error {
+	if !vp.InConfig(key) {
+		return fmt.Errorf("unknown key: %s", key)
+	}
+
+	// each viper key/val entry should be bound to a command flag
+	flag := cmd.Flag(key)
+	if flag == nil {
+		return fmt.Errorf("key=%s not bound to a flag", key)
+	}
+
+	val := value
+	if value == "" {
+		val = flag.DefValue
+	}
+
+	var err error
+	var newVal interface{}
+	typ := flag.Value.Type()
```

Comment:
Looks like viper also has `flag.ValueType()`. Unclear if it's any different than what you wrote

https://github.com/cilium/hubble/blob/master/vendor/github.com/spf13/viper/viper.go#L1053

Reply:
It looks like it's [literally](https://github.com/cilium/hubble/blob/2b8c000824cdfc9ff9fb64d9bac408de795f9c61/vendor/github.com/spf13/viper/flags.go#L54-L57) the same:
```go
// ValueType returns the type of the flag as a string.
func (p pflagValue) ValueType() string {
	return p.flag.Value.Type()
}
```